cmd/spire-server/cli/entry/update.go

```diff
@@ -0,0 +1,210 @@
+package entry
+
+import (
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/spiffe/spire/cmd/spire-server/util"
+	"github.com/spiffe/spire/pkg/common/idutil"
+	"github.com/spiffe/spire/proto/api/registration"
+	"github.com/spiffe/spire/proto/common"
+
+	"golang.org/x/net/context"
+)
+
+type UpdateConfig struct {
+	// Socket path of registration API
+	RegistrationUDSPath string
+
+	// Path to an optional data file. If set, other
+	// opts will be ignored.
+	Path string
+
+	// Registration entry id to update
+	EntryID string
+
+	// Type and value are delimited by a colon (:)
+	// ex. "unix:uid:1000" or "spiffe_id:spiffe://example.org/foo"
+	Selectors StringsFlag
+
+	ParentID string
+	SpiffeID string
+	Ttl      int
+
+	// List of SPIFFE IDs of trust domains the registration entry is federated with
+	FederatesWith StringsFlag
+}
+
+// Perform basic validation, even on fields that we
+// have defaults defined for
+func (rc *UpdateConfig) Validate() (err error) {
```

Comment:
It seems that this will validate all the fields but will not return an error if an entry ID is not specified. So looks like the rpc will be called and fail with an error due to a missing entry ID, which may be confusing.
We may just check for `rc.EntryID == ""` as the first thing here and return an error with something like `"missing registration entry id to update"`?

Reply:
good catch! updated.